### PR TITLE
Use LiveKit's Rust SDK on Linux while continue using Swift SDK on Mac

### LINKED ISSUE #13714: Add support for LiveKit and mic/screensharing on Linux

### PR DESCRIPTION
**Pull request by mgsloan**
Similar to #20826 but keeps the Swift implementation.  There were quite a few changes in the `call` crate, and so that code now has two variants.

Closes #13714

Release Notes:

- Added preliminary Linux support for voice chat and viewing screenshares.

### CODE REVIEW

File: crates/collab/k8s/collab.template.yml [hunk 1]
```diff
@@ -109,17 +109,17 @@ spec:
...
                   name: zed-client
                   key: checksum-seed
-            - name: LIVE_KIT_SERVER
+            - name: LIVEKIT_SERVER
               valueFrom:
                 secretKeyRef:
                   name: livekit
                   key: server
-            - name: LIVE_KIT_KEY
+            - name: livekit_KEY
```
Comment (maxdeviant) [inline crates/collab/k8s/collab.template.yml:117]: This environment variable name looks incorrect.

Comment (mgsloan) [reply crates/collab/k8s/collab.template.yml:117]: Good catch!

**Comment by WeetHet** [comment]
Sorry to interfere, before this is merged, could we get a feature to build macOS with the Rust SDK? I understand that it might be difficult to implement, but it would greatly reduce the difficulty in packaging Zed in nixpkgs, as we only have swift 5.8 now. It would be greatly appreciated

**Comment by mikayla-maki** [comment]
@WeetHet We just pushed a commit that adds a `livekit-cross-platform` feature to the `workspace`. You should be able to enable it in nix by adjusting a few lines here: https://github.com/zed-industries/zed/pull/21550/files#diff-315ed179e69e42c0a098166a7faa7899838376c2649c84fa8cb66643f3813a45R129-R134

**Comment by WeetHet** [comment]
> @WeetHet We just pushed a commit that adds a `livekit-cross-platform` feature to the `workspace`. Once we've fixed CI, you should be able to enable the cross platform livekit code in nixpkgs by adjusting a few lines here: https://github.com/zed-industries/zed/pull/21550/files#diff-315ed179e69e42c0a098166a7faa7899838376c2649c84fa8cb66643f3813a45R129-R134
> 
> Unfortunately, due to the way platform dependencies work, I don't know if we can make this configurable by the CLI

Thank you, that's probably more than enough

**Comment by huacnlee** [comment]
After this changes, Zed can't work on macOS 11.x.

```
dyld: Library not loaded: /System/Library/Frameworks/ScreenCaptureKit.framework/Versions/A/ScreenCaptureKit
```

Ref issue #22419

**Pull request merged**